### PR TITLE
[Frameworks] HuggingFace model server

### PR DESCRIPTION
**Pull request by yonishelach**
Built-in support for the HuggingFace framework - the transformers hub:
* Added a serving class for HuggingFace `pipeline` which is constructed from a `model` and a `tokenizer`.
* Added tests for serving.
* Updated `dev-requirements.txt` with the HuggingFace package - `transformers`

### CODE REVIEW

File: mlrun/frameworks/huggingface/model_server.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+            tokenizer_object = getattr(transformers, self.tokenizer_class)
+            self.tokenizer = tokenizer_object.from_pretrained(self.tokenizer_name)
+
+        # Initializing the pipeline:
+        self.pipe = transformers.pipeline(
+            task=self.task,
+            model=self.model or self.model_name,
+            tokenizer=self.tokenizer,
+            framework=self.framework,
+        )
```
Comment (guy1992l) [inline mlrun/frameworks/huggingface/model_server.py:106]: We will later want to have a `HuggingFaceModelHandler` here to load the model.

File: mlrun/frameworks/huggingface/model_server.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+        # Loading the pretrained tokenizer:
+        if self.tokenizer_class:
+            tokenizer_object = getattr(transformers, self.tokenizer_class)
+            self.tokenizer = tokenizer_object.from_pretrained(self.tokenizer_name)
+
+        # Initializing the pipeline:
+        self.pipe = transformers.pipeline(
+            task=self.task,
+            model=self.model or self.model_name,
```
Comment (guy1992l) [inline mlrun/frameworks/huggingface/model_server.py:103]: why `or self._model_name` ?

File: tests/frameworks/huggingface/serving.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        " and province of Île-de-France, or Paris Region, which has an estimated population of 12,174,880,"
+        " or about 18 percent of the population of France as of 2017.",
+        "result_keys": ["summary_text"],
+    },
+    {
+        "task": "translation_en_to_fr",
+        "example": "How old are you?",
+        "result_keys": ["translation_text"],
+    },
+]
```
Comment (guy1992l) [inline tests/frameworks/huggingface/serving.py:78]: Looks like a copy-paste from HuggingFace, I'm not sure it is a good thing. Please consult with Yaron if he approves.

**Review by guy1992l — COMMENTED** [review]
It is a very specific server case. It should be more configurable. For example:
* If I would like to make the tokenizer as separate step in the serving graph I don't want it to be loaded automatically into a pipeline.
* If I trained a model using HuggingFace and saved it, the model name argument shouldn't be a hub link, it could be simply my model...

**Comment by Tankilevitch** [comment]
Duplicate PR with #2594 , closing

**Pull request closed without merging**